### PR TITLE
refactor: change signature for `tuiGetElementObscures`

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/cdk/utils/dom/get-element-obscurers.ts [hunk 1]
```diff
@@ -5,14 +5,10 @@ import {tuiGetDocumentOrShadowRoot} from './get-document-or-shadow-root';
 
 /**
  * Returns array of Elements covering edges of given element or null if at least one edge middle point is visible
- *
- * CAUTION: Empty array means element if offscreen i.e. covered by no elements, rather than not covered
```
Comment (nsbarsukov) [inline projects/cdk/utils/dom/get-element-obscurers.ts:9]: Why was this line deleted ?

File: projects/cdk/utils/dom/get-element-obscurers.ts [hunk 2]
```diff
@@ -50,10 +46,12 @@ export function tuiGetElementObscures(element: Element): readonly Element[] | nu
     const nonNull = elements.filter(tuiIsPresent);
 
     if (!nonNull.length) {
-        return nonNull;
+        return nonNull as [Element, Element, Element, Element];
```
Comment (nsbarsukov) [inline projects/cdk/utils/dom/get-element-obscurers.ts:49]:
```suggestion
        return nonNull as [];
```

Comment (splincode) [reply projects/cdk/utils/dom/get-element-obscurers.ts]: fixed

File: projects/cdk/utils/dom/get-element-obscurers.ts [hunk 3]
```diff
@@ -50,10 +51,12 @@ export function tuiGetElementObscures(element: Element): readonly Element[] | nu
     const nonNull = elements.filter(tuiIsPresent);
 
     if (!nonNull.length) {
-        return nonNull;
+        return nonNull as [];
```
Comment (waterplea) [inline projects/cdk/utils/dom/get-element-obscurers.ts:54]: No need to typecast, we can just return `[]`

**Review by waterplea — APPROVED** [review]

File: projects/cdk/utils/dom/get-element-obscurers.ts (hunk 3 above)
Comment (splincode) [reply projects/cdk/utils/dom/get-element-obscurers.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]
